test/topicFilterCopy.spec.ts

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import chai from "chai";
+const should = chai.should();
+import chaiAsPromised from "chai-as-promised";
+import dotenv from "dotenv";
+dotenv.config();
+chai.use(chaiAsPromised);
+
+import {
+  Namespace,
+  SubscriptionClient,
+  SendableMessageInfo,
+  TopicClient,
+  ServiceBusMessage,
+  delay
+} from "../lib";
+
+// We need to remove rules before adding one because otherwise the existing default rule will let in all messages.
+async function removeAllRules(client: SubscriptionClient): Promise<void> {
+  const rules = await client.getRules();
+  for (let i = 0; i < rules.length; i++) {
+    const rule = rules[i];
+    await client.removeRule(rule.name);
+  }
+}
+
+async function testPeekMsgsLength(
+  client: SubscriptionClient,
+  expectedPeekLength: number
+): Promise<void> {
+  const peekedMsgs = await client.peek(expectedPeekLength + 1);
+  should.equal(
+    peekedMsgs.length,
+    expectedPeekLength,
+    "Unexpected number of msgs found when peeking"
+  );
+}
+
+let namespace: Namespace;
+let topicClient: TopicClient;
+let subscriptionClient: SubscriptionClient;
+let defaultSubscriptionClient: SubscriptionClient;
+
+async function beforeEachTest(): Promise<void> {
+  // The tests in this file expect the env variables to contain the connection string and
+  // the names of empty queue/topic/subscription that are to be tested
+
+  if (!process.env.SERVICEBUS_CONNECTION_STRING) {
+    throw new Error(
+      "Define SERVICEBUS_CONNECTION_STRING in your environment before running integration tests."
+    );
+  }
+  if (!process.env.TOPIC_NAME) {
+    throw new Error("Define TOPIC_NAME in your environment before running integration tests.");
+  }
+  if (!process.env.SUBSCRIPTION_NAME) {
+    throw new Error(
+      "Define SUBSCRIPTION_NAME in your environment before running integration tests."
+    );
+  }
+  if (!process.env.DEFAULT_SUBSCRIPTION_NAME) {
+    throw new Error(
+      "Define DEFAULT_SUBSCRIPTION_NAME in your environment before running integration tests."
+    );
+  }
+
+  namespace = Namespace.createFromConnectionString(process.env.SERVICEBUS_CONNECTION_STRING);
+  topicClient = namespace.createTopicClient(process.env.TOPIC_NAME);
+  subscriptionClient = namespace.createSubscriptionClient(
+    process.env.TOPIC_NAME,
+    process.env.SUBSCRIPTION_NAME
+  );
+  defaultSubscriptionClient = namespace.createSubscriptionClient(
+    process.env.TOPIC_NAME,
+    process.env.DEFAULT_SUBSCRIPTION_NAME
+  );
+  await removeAllRules(subscriptionClient);
+}
+
+async function afterEachTest(): Promise<void> {
+  await namespace.close();
+}
+
+async function sendOrder(order: any): Promise<void> {
```

Comment:
This function is only being called from `sendOrders()`. Can we merge the 2 functions?